--- origsrc/deja-dup-28.0/deja-dup/nautilus/Makefile.am	2013-09-10 00:37:00.000000000 -0500
+++ src/deja-dup-28.0/deja-dup/nautilus/Makefile.am	2014-02-28 11:46:47.149261400 -0600
@@ -30,6 +30,6 @@ libdeja_dup_la_CFLAGS = \
 libdeja_dup_la_SOURCES = \
 	NautilusExtension.c NautilusExtension.h
 
-libdeja_dup_la_LDFLAGS = -module -avoid-version
+libdeja_dup_la_LDFLAGS = -module -avoid-version -no-undefined
 libdeja_dup_la_LIBADD = $(NAUTILUS_LIBS) @INTLLIBS@
 endif
--- origsrc/deja-dup-28.0/deja-dup/widgets/Makefile.am	2013-09-10 00:34:44.000000000 -0500
+++ src/deja-dup-28.0/deja-dup/widgets/Makefile.am	2014-02-28 12:08:44.479608400 -0600
@@ -25,7 +25,7 @@ libwidgets_la_CFLAGS = \
 	-w \
 	-DGETTEXT_PACKAGE=\"$(GETTEXT_PACKAGE)\"
 
-libwidgets_la_LDFLAGS = -avoid-version
+libwidgets_la_LDFLAGS = -avoid-version -no-undefined -release $(PACKAGE)
 
 libwidgets_la_LIBADD = \
 	$(WIDGETS_LIBS) \
--- origsrc/deja-dup-28.0/libdeja/Makefile.am	2013-09-10 00:46:07.000000000 -0500
+++ src/deja-dup-28.0/libdeja/Makefile.am	2014-02-28 11:46:08.290038800 -0600
@@ -26,7 +26,7 @@ libdeja_la_CFLAGS = \
 	-DPKG_LIBEXEC_DIR=\"$(pkglibexecdir)\" \
 	-DGETTEXT_PACKAGE=\"$(GETTEXT_PACKAGE)\"
 
-libdeja_la_LDFLAGS = -avoid-version
+libdeja_la_LDFLAGS = -avoid-version -no-undefined
 
 libdeja_la_LIBADD = \
 	$(LIBDEJA_LIBS) \
--- origsrc/deja-dup-28.0/libdeja/tools/duplicity/Makefile.am	2013-09-10 00:40:10.000000000 -0500
+++ src/deja-dup-28.0/libdeja/tools/duplicity/Makefile.am	2014-02-28 11:50:14.649129700 -0600
@@ -27,7 +27,7 @@ libduplicity_la_CFLAGS = \
 	-DPKG_LIBEXEC_DIR=\"$(pkglibexecdir)\" \
 	-DGETTEXT_PACKAGE=\"$(GETTEXT_PACKAGE)\"
 
-libduplicity_la_LDFLAGS = -module -avoid-version
+libduplicity_la_LDFLAGS = -module -avoid-version -no-undefined
 
 libduplicity_la_LIBADD = \
 	$(LIBDEJA_LIBS) \
