--- origsrc/deja-dup-32.0/CMakeLists.txt	2014-09-20 10:04:47.000000000 -0500
+++ src/deja-dup-32.0/CMakeLists.txt	2015-06-24 17:24:57.396380700 -0500
@@ -25,6 +25,7 @@ include(DejaDupMacros)
 
 deja_option(ENABLE_CCPANEL "Build gnome-control-center panel" CHECK)
 deja_option(ENABLE_NAUTILUS "Build nautilus plugin" CHECK)
+deja_option(ENABLE_CAJA "Build caja plugin" CHECK)
 deja_option(ENABLE_UNITY "Integrate with Unity" CHECK)
 deja_option(ENABLE_UNITY_CCPANEL "Build unity-control-center panel" CHECK)
 
@@ -53,6 +54,14 @@ if(ENABLE_NAUTILUS)
                   OUTPUT_STRIP_TRAILING_WHITESPACE
                   ERROR_QUIET)
 endif()
+
+deja_enable_option(CAJA libcaja-extension)
+if(ENABLE_CAJA)
+  execute_process(COMMAND ${PKG_CONFIG_EXECUTABLE} --variable=extensiondir libcaja-extension
+                  OUTPUT_VARIABLE CAJA_EXTENSIONDIR
+                  OUTPUT_STRIP_TRAILING_WHITESPACE
+                  ERROR_QUIET)
+endif()
 
 deja_enable_option(UNITY_CCPANEL libunity-control-center)
 if(ENABLE_UNITY_CCPANEL)
--- origsrc/deja-dup-32.0/deja-dup/CMakeLists.txt	2014-05-02 11:47:15.000000000 -0500
+++ src/deja-dup-32.0/deja-dup/CMakeLists.txt	2015-06-24 17:21:33.982746100 -0500
@@ -50,6 +50,7 @@ install(DIRECTORY ui DESTINATION "${PKG_
 add_subdirectory(help)
 add_subdirectory(monitor)
 add_subdirectory(nautilus)
+add_subdirectory(caja)
 add_subdirectory(preferences)
 add_subdirectory(tests)
 add_subdirectory(widgets)
